From: Richard Acayan <mailingradian@gmail.com>
Date: Fri, 2 Aug 2024 22:09:39 -0400
Subject: [PATCH] dt-bindings: media: i2c: Add Sony IMX355

The IMX355 camera sensor is a camera sensor that can be found as the
front camera in some smartphones, such as the Pixel 3, Pixel 3 XL, Pixel
3a, and Pixel 3a XL. It already has a driver, but needs support for
device tree. Document the IMX355 to support defining it in device tree.

Signed-off-by: Richard Acayan <mailingradian@gmail.com>
---
 .../devicetree/bindings/media/i2c/sony,imx355.yaml | 117 +++++++++++++++++++++
 1 file changed, 117 insertions(+)
 create mode 100644 Documentation/devicetree/bindings/media/i2c/sony,imx355.yaml

diff --git a/Documentation/devicetree/bindings/media/i2c/sony,imx355.yaml b/Documentation/devicetree/bindings/media/i2c/sony,imx355.yaml
new file mode 100644
index 0000000..6c3fbfb
--- /dev/null
+++ b/Documentation/devicetree/bindings/media/i2c/sony,imx355.yaml
@@ -0,0 +1,117 @@
+# SPDX-License-Identifier: (GPL-2.0-only OR BSD-2-Clause)
+
+%YAML 1.2
+---
+$id: http://devicetree.org/schemas/media/i2c/sony,imx355.yaml#
+$schema: http://devicetree.org/meta-schemas/core.yaml#
+
+title: Sony IMX355 Sensor
+
+maintainers:
+  - Richard Acayan <mailingradian@gmail.com>
+
+description:
+  The IMX355 sensor is a 3280x2464 image sensor, commonly found as the front
+  camera in smartphones.
+
+allOf:
+  - $ref: /schemas/media/video-interface-devices.yaml#
+
+properties:
+  compatible:
+    const: sony,imx355
+
+  reg:
+    maxItems: 1
+
+  assigned-clocks: true
+  assigned-clock-rates: true
+
+  clocks:
+    maxItems: 1
+
+  clock-names:
+    items:
+      - const: mclk
+
+  clock-frequency:
+    $ref: /schemas/types.yaml#/definitions/uint32
+    description: External clock frequency.
+
+  vana-supply:
+    $ref: /schemas/types.yaml#/definitions/phandle
+    description: Analog power supply.
+
+  vdig-supply:
+    $ref: /schemas/types.yaml#/definitions/phandle
+    description: Analog power supply.
+
+  vio-supply:
+    $ref: /schemas/types.yaml#/definitions/phandle
+    description: Interface power supply.
+
+  reset-gpios:
+    maxItems: 1
+
+  port:
+    $ref: /schemas/graph.yaml#/$defs/port-base
+    description:
+      CSI output port.
+
+    properties:
+      endpoint:
+        $ref: /schemas/media/video-interfaces.yaml
+        description:
+          CSI endpoint for the sensor output.
+
+        unevaluatedProperties: false
+
+        required:
+          - link-frequencies
+
+    unevaluatedProperties: false
+
+    required:
+      - endpoint
+
+unevaluatedProperties: false
+
+required:
+  - compatible
+  - reg
+  - clock-frequency
+
+examples:
+  - |
+    camera@1a {
+        compatible = "sony,imx355";
+        reg = <0x1a>;
+
+        clocks = <&camcc CAM_CC_MCLK2_CLK>;
+        clock-names = "mclk";
+
+        clock-frequency = <19200000>;
+
+        assigned-clocks = <&camcc CAM_CC_MCLK2_CLK>;
+        assigned-clock-rates = <24000000>;
+
+        reset-gpios = <&tlmm 9 GPIO_ACTIVE_HIGH>;
+
+        vana-supply = <&cam_front_ldo>;
+        vdig-supply = <&cam_front_ldo>;
+        vio-supply = <&cam_vio_ldo>;
+
+        pinctrl-names = "default";
+        pinctrl-0 = <&cam_front_default>;
+
+        rotation = <270>;
+        orientation = <0>;
+
+        port {
+            cam_front_endpoint: endpoint {
+                data-lanes = <0 1 2 3>;
+                link-frequencies = /bits/ 64 <360000000>;
+                remote-endpoint = <&camss_endpoint1>;
+            };
+        };
+    };
